src/content/blog/2024_zne_catalyst.md

```diff
@@ -0,0 +1,44 @@
+---
+title: "Just in time for just-in-time error mitigation"
+author: Alessandro Cosentino
+day: 31
+month: 10
+year: 2024
+---
+
+We all want faster computer programs. Just-in-time (JIT) compilation has been a key concept in classical compiler design to help achieve faster computer programs. Most modern programming languages and runtime environments support some kind of JIT optimization techniques. In fact, very recently, an [experimental JIT compiler](https://docs.python.org/3/whatsnew/3.13.html#whatsnew313-jit-compiler) was added to Python in its latest release, 3.13.
+
+Just as JIT compilation accelerates classical programs, similar techniques can be applied to quantum computing to boost performance. We all want faster quantum computer programs.  Catalyst is a JIT compiler framework developed by Xanadu to speed up quantum programs written with their PennyLane library. If you have a PennyLane program, you can decorate it with some simple Catalyst syntax and, in many cases, you have a faster quantum computer program.
+
+What we also want is more accurate quantum computer programs – accurate enough to be useful for the application or the experiment we seek to run. Quantum error correction is making significant strides, but it still introduces substantial overhead that can be impractical for near-term applications. We look at the more practical techniques of quantum error mitigation to make our quantum programs more accurate. Unitary Fund's own Mitiq project provides a Python package for applying error mitigation techniques to quantum programs. 
+
+Even though Mitiq can be used with PennyLane programs, it doesn't seamless integrate into the Catalyst framework for JIT compilation. In the last few months, Unitary Fund and Xanadu have partnered to improve the integration of error mitigation techniques into the Catalyst compiler. The natural starting point was Zero-Noise Extrapolation (ZNE), which is
```

Comment:
```suggestion
Even though [Mitiq](https://github.com/unitaryfund/mitiq) could already be used with PennyLane programs, it didn't seamlessly integrate into the Catalyst framework for JIT compilation. In the last few months, Unitary Fund and Xanadu have partnered to improve the integration of error mitigation techniques, upstreaming them directly from Mitiq into the Catalyst compiler. The natural starting point was Zero-Noise Extrapolation (ZNE), which is
```

Reply:
this is the first time Mitiq is cited, in the blog post, I think it should be introduced in a previous line.

Reply:
It's not true that now Mitiq itself integrates into Catalyst, that's why I didn't use the past tense. It's also not true that we upstreamed techniques directly from Mitiq. What we did is rewriting those techniques from scratch in the Catalyst framework.
This has been a matter of confusion since the initial proposal, and it's something I have highlighted in the final report as well.